src/WebJobs.Script/Workers/Profiles/WorkerProfileManager.cs

```diff
@@ -14,17 +14,30 @@ namespace Microsoft.Azure.WebJobs.Script.Workers
     /// </summary>
     internal class WorkerProfileManager : IWorkerProfileManager
     {
+        private const string LogCategory = "Host.WorkerProfiles";
+
         private readonly ILogger _logger;
+        private readonly IEnvironment _environment;
         private readonly IEnumerable<IWorkerProfileConditionProvider> _conditionProviders;
+
+        private string _activeProfile = string.Empty;
         private Dictionary<string, List<WorkerDescriptionProfile>> _profiles;
-        private string _activeProfile;
 
-        public WorkerProfileManager(ILogger logger, IEnumerable<IWorkerProfileConditionProvider> conditionProviders)
+        public WorkerProfileManager(ILoggerFactory loggerFactory, IEnvironment environment)
         {
-            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            _conditionProviders = conditionProviders ?? throw new ArgumentNullException(nameof(conditionProviders));
+            if (loggerFactory is null)
```

Comment:
Are we on a new enough .NET version to use `ArgumentNullException.ThrowIfNull(loggerFactory)`?